sites/cheerpj/src/content/docs/13-tutorials/05-interoperability-tutorial.mdx

```diff
@@ -0,0 +1,193 @@
+
+You can see the fully implemented example on [GitHub](https://github.com/leaningtech/cheerpj-meta/tree/main/examples/Interoperability).
+
+## Java Source Code: `Example.java`
+
+Let's start by examining the source code of the Java class that will interact with JavaScript. This class will receive input from JavaScript, process it, and send a response back to JavaScript.
+
+```java title="Example.java"
+package com.example;
+
+public class Example {
+    public static native void sendToHTML(String s);
+
+    public String processInput(String input) {
+        sendToHTML(input);
+        return "Java received: " + input;
+    }
+
+    public static native void nativeSetApplication(Example myApplication);
+
+    public static void main(String[] args) {
+        Example app = new Example();
+        new Thread(() -> {
+            nativeSetApplication(app);
+            System.out.println("Starting Thread");
+        }).start();
+    }
+}
+```
+
+This Java class contains the following methods:
+
+- `sendToHTML`: A native method implemented in JavaScript, called to send data to the JavaScript environment.
+- `processInput`: Receives data from JavaScript, passes it to `sendToHTML`, and returns a response.
+- `nativeSetApplication`: Captures the running Java thread for persistent communication with JavaScript.
+
+## JavaScript Native Method Implementation
+
+To enable Java and JavaScript communication, we need to implement the [`native`] methods in JavaScript. These methods will handle the data exchange between the two environments.
+
+We will do this in the `index.html` file, which is part of the template project `.zip` file. In this file, you will find useful comments to guide you through the implementation process described below.
+
+### 1. Define JavaScript Native Methods
+
+In the `<script>` tag, implement the native methods for interacting with Java.
+
+To implement the native methods in JavaScript, we create an `async` function for each Java method following the naming convention `Java_<fully-qualified-class-name>_<method-name>`. The function should accept `lib` as the first parameter and the Java instance as the second parameter (if applicable). See the [`native method`] implementation guide for more details.
+
+You can move these methods to a separate `.js` file for better organization and include them using the `<script>` tag in the HTML.
+
+```js title="index.html"
+async function Java_com_example_Example_sendToHTML(lib, str) {
+	document.getElementById("javaOutput").innerText =
+		"JavaScript received: " + str;
+	console.log("Received input from Java: " + str);
+}
+
+async function Java_com_example_Example_nativeSetApplication(
+	lib,
+	myApplication
+) {
+	window.myApplication = myApplication;
+	console.log("Java application instance set on JavaScript side.");
+	return new Promise(() => {}); // Keeps the function from returning
+}
+```
```

Comment:
You forgot to add this part here, currently the inputDiv in the index.html file stays hidden.
`document.getElementById("inputDiv").style.display = "unset";`